src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -154,18 +154,18 @@ void System.Collections.ICollection.CopyTo(Array array, int index)
 
             if (array.Rank != 1)
             {
-                throw new ArgumentException(SR.Arg_RankMultiDimNotSupported);
+                throw new ArgumentException(SR.Arg_RankMultiDimNotSupported, "array");
             }
 
             if (array.GetLowerBound(0) != 0)
             {
-                throw new ArgumentException(SR.Arg_NonZeroLowerBound);
+                throw new ArgumentException(SR.Arg_NonZeroLowerBound, "array");
             }
 
             int arrayLen = array.Length;
-            if (index < 0 || index > arrayLen)
+            if ((uint)index > (uint)arrayLen)
```

Comment:
Ditto on why we're changing these.... I realize it's done like this elsewhere in the code, but I still find the original more easily understood.